docs/two-phase-commit-transactions.md

```diff
@@ -1,33 +1,27 @@
-# Two-phase Commit Transactions
+# Two-Phase Commit Transactions
```

Comment:
@feeblefakie So, should we also rename it to `Transactions with a two-phase interface`?

Reply:
@brfrn169 Yes. @feeblefakie reminded me of the following comment from a different PR: https://github.com/scalar-labs/scalardb/pull/934#issuecomment-1644890585

With that in mind, I've changed mentions of `two-phase commit transactions` to `transactions with a two-phase commit interface` in this PR. I've also noted this wording in the Vale style guide that I currently reference on my side and will change it on docs moving forward.